docs/bytecode.md

```diff
@@ -0,0 +1,208 @@
+# StableHLO Bytecode
+
+## Basic Example
+A minimalist example which only encodes a single attribute in StableHLO can be found [at this commit](https://github.com/openxla/stablehlo/commit/00e3dc98d8d956e5e494be3022df973821e58e91).
+
+There is an image at the bottom of the page which shows difference in the bianry file.
+
+## Currently Encoded Attributes / Types
+
+### Attributes
+
+```
+ArgResultAlias {
+  argTupleIndices: svarint[]
+  resultIndex: svarint
+  resultIndex: svarint[]
+  isMustAlias: varint
+}
+
+ChannelHandleAttr {
+  handle: svarint
+  type: svarint
+}
+
+ComparisonDirectionAttr
+  ComparisonDirection: varint
+}
+
+ComparisonTypeAttr
+  ComparisonType: varint
+}
+
+ConvDimensionNumbersAttr {
+  inputBatchDimension: svarint
+  inputFeatureDimension: svarint
+  inputSpatialDimensions: svarint[]
+  kernelInputFeatureDimension: svarint
+  kernelOutputFeatureDimension: svarint
+  kernelSpatialDimensions: svarint[]
+  outputBatchDimension: svarint
+  outputFeatureDimension: svarint
+  outputSpatialDimensions: svarint[]
+}
```

Comment:
Yeah so slight edge case on custom call api verision, unclear if this is intentional or not. This enum only defined as

Note the following line for `FftType`:
https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/StablehloEnums.td#L72
```
def StableHLO_FftTypeAttr : EnumAttr<StableHLO_Dialect, StableHLO_FftType, "fft_type">;
```

There is no corresponding declaration that ties `CustomCallApiVersion` to the StableHLO dialect, so it is only defined/used as a `Int32EnumAttr` and therefore handled by the builtin encoding it seems.

I added the following note to `bytecode.md`:

```
Special Cases:
- `StableHLO_ConvolutionAttributes`
  + Despite its name,  is not an attribute and is not encoded.
    Rather, it is a dag which gets expanded into several attributes
    which are all encoded separately.
- `StableHLO_CustomCallApiVersionAttr`
  + This enum is defined strictly as an attribute of `I32EnumAttr`
    and not an `EnumAttr` of the `StablehloDialect`. This differs from
   `FftType` and other enum attributes. Because of this, it is handled by
    the builtin encoding.
```

Reply:
Hmm I don't think that you pushed this update?

Reply:
Huh.. where did that go.. let me double check, may have been on the wrong branch making that edit